lib/Dialect/HW/Transforms/FlattenModules.cpp

```diff
@@ -118,13 +248,37 @@ void FlattenModulesPass::runOnOperation() {
       if (numUsesLeft == 0)
         continue;
 
-      for (auto *instRecord : node->uses()) {
-        // Only inline private `HWModuleOp`s (no extern or generated modules).
-        auto module =
-            dyn_cast_or_null<HWModuleOp>(node->getModule().getOperation());
-        if (!module || !module.isPrivate())
-          continue;
+      // Only inline private `HWModuleOp`s (no extern or generated modules).
+      auto module =
+          dyn_cast_or_null<HWModuleOp>(node->getModule().getOperation());
+      if (!module || !module.isPrivate())
+        continue;
+
+      // Do not inline a module if it is targeted by a module NLA.
+      if (leafModules.count(module.getNameAttr()))
+        continue;
+
+      // Check if module should be inlined based on heuristics
+      auto *body = module.getBodyBlock();
+      size_t bodySize = std::distance(body->begin(), body->end());
```

Comment:
Okay, I think this is number of ops, right, as this is iterator distance? That seems to make sense.

Reply:
Right, its the number of ops.